tests/lib/mem_alloc/prj_negative_testing.conf

```diff
@@ -1,3 +1,5 @@
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
+CONFIG_NO_OPTIMIZATIONS=y
```

Comment:
There is a big difference between disabling optimisation globally vs. only for some functions of interest.

Reply:
Agreed, was trying a few things while testing and forgot to remove them.